### DISCUSSION TITLE
How to Project And Draw the detected lane lines?

### DISCUSSION THREAD

**Discussion by satyamjay-iitd** [thomasfermi/Algorithms-for-Automated-Driving#28] [General]
I want to draw lane lines in some other camera frame. The equation of the detected line is in the "Lane Detection Camera" frame, right?

My initial idea was to:-
- Find some (x,y,z) points in "Lane Detection Camera" space that belongs to the lane-line using the line equation.
- Transform them in any other camera space.
- Project them on the 2d canvas, and draw them using pygame.draw api.

But I am stuck at the first step. Any ideas/suggestion?

**Comment by thomasfermi** [comment]
Hi @satyamjay-iitd, I am not sure I understand your problem. You have two cameras c1 and c2. You want to detect the lane lines in the images from camera c1, and then you want to draw them into the images from camera c2. Is that right?

**Reply by satyamjay-iitd** [reply]
Yes exactly

**Reply by thomasfermi** [reply]
I think that your 3 step idea should work. Regarding the first step: That is actually exactly what is being discussed in the first chapter of the book:

https://thomasfermi.github.io/Algorithms-for-Automated-Driving/LaneDetection/LaneDetectionOverview.html
